commands/new_function.go

```diff
@@ -81,12 +85,27 @@ the "Dockerfile" lang type in your YAML file.
 		return fmt.Errorf("you must supply a function language with the --lang flag")
 	}
 
-	PullTemplates("")
+	existingYaml := len(yamlFile) > 0
+	var stackFileName string
+	if !existingYaml {
+		// We will create a new YAML file for this function
+		stackFileName = functionName + ".yml"
+	} else {
+		stackFileName = yamlFile
+		// YAML file was passed in, so parse to see if it is valid
+		parsedServices, err := stack.ParseYAMLFile(stackFileName, "", "")
+		if err != nil || parsedServices == nil {
+			return fmt.Errorf("Specified file (" + stackFileName + ") is not valid YAML\n")
+		}
+		//TODO: Verify that services is good
```

Comment:
What were you intending to do here? If the PR is complete as-is then remove this comment, if its a non-blocking enhancement then raise as a separate issue if not implementing now.